pkg/controllers/management/eks/eks_cluster_handler.go

```diff
@@ -171,9 +172,13 @@ func (e *eksOperatorController) onClusterChange(key string, cluster *mgmtv3.Clus
 			return cluster, err
 		}
 
-		logrus.Infof("waiting for EKS cluster [%s] to finish creating", cluster.Name)
 		e.clusterEnqueueAfter(cluster.Name, 20*time.Second)
-		return cluster, nil
+		if failureMessage == "" {
+			logrus.Infof("waiting for cluster EKS [%s] to finish creating", cluster.Name)
+			return e.setUnknown(cluster, apimgmtv3.ClusterConditionProvisioned, "")
```

Comment:
why is there a difference between the provisioned condition being set to unknown here vs false if the message exists?